docs/advanced/server-side-rendering.md

```diff
@@ -177,254 +177,156 @@ we'll have to import from. `preact-ssr-prepass`.
-more involved, but has two major benefits:
+import { UrqlProvider, ssrExchange, cacheExchange, fetchExchange, createClient } from '@urql/next';
 
-1. allows **direct schema execution** for performance optimisation
-2. allows performing extra operations in those functions
-
-To make the functions work with the `withUrqlClient` wrapper, return the `urqlState` prop
-with the extracted data from the `ssrExchange`:
-
-```js
-import { withUrqlClient, initUrqlClient } from 'next-urql';
-import { ssrExchange, dedupExchange, cacheExchange, fetchExchange, useQuery } from 'urql';
-
-const TODOS_QUERY = `
-  query { todos { id text } }
-`;
+const ssr = ssrExchange();
+const client = createClient({
+  url: 'https://trygql.formidable.dev/graphql/web-collections',
+  exchanges: [cacheExchange, ssr, fetchExchange],
+});
 
-function Todos() {
-  const [res] = useQuery({ query: TODOS_QUERY });
+export default function Layout({ children }: React.PropsWithChildren) {
   return (
-    <div>
-      {res.data.todos.map(todo => (
-        <div key={todo.id}>
-          {todo.id} - {todo.text}
-        </div>
-      ))}
-    </div>
+    <UrqlProvider client={client} ssr={ssr}>
+      {children}
+    </UrqlProvider>
   );
 }
-
-export async function getStaticProps(ctx) {
-  const ssrCache = ssrExchange({ isClient: false });
-  const client = initUrqlClient(
-    {
-      url: 'your-url',
-      exchanges: [dedupExchange, cacheExchange, ssrCache, fetchExchange],
-    },
-    false
-  );
-
-  // This query is used to populate the cache for the query
-  // used on this page.
-  await client.query(TODOS_QUERY).toPromise();
-
-  return {
-    props: {
-      // urqlState is a keyword here so withUrqlClient can pick it up.
-      urqlState: ssrCache.extractData(),
-    },
-    revalidate: 600,
-  };
-}
-
-export default withUrqlClient(
-  ssr => ({
-    url: 'your-url',
-  })
-  // Cannot specify { ssr: true } here so we don't wrap our component in getInitialProps
-)(Todos);
 ```
 
-The above example will make sure the page is rendered as a static-page, It's important that
-you fully pre-populate your cache so in our case we were only interested in getting our todos,
-if there are child components relying on data you'll have to make sure these are fetched as well.
+It is important that we pas both a client as well as the `ssrExchange` to the `Provider`
+this way we will be able to restore the data that Next streams to the client later on
+when we are hydrating.
 
-The `getServerSideProps` and `getStaticProps` functions only run on the **server-side** — any
-code used in them is automatically stripped away from the client-side bundle using the
-[next-code-elimination tool](https://next-code-elimination.vercel.app/). This allows **executing
-our schema directly** using `@urql/exchange-execute` if we have access to our GraphQL server:
+The next step is to query data in your client components by means of the `useQuery`
+method defined in `@urql/next`.
```

Comment:
does it _have_ to import them from `@urql/next` rather than the base urql package? I'd prefer to have it import the base one to make the component re-usable in contexts outside of Next.js (like React Native).

Reply:
This is a valid point, it's pretty hard to circumvent this however as we need a trigger to check the data being streamed to the client. i.e. when a suspense boundary resolves on the server and hydration continues it will naturally call `useQuery` else we would need an additional hook. Maybe there are ways to hook into next/React telling us more data is available but I haven't found it yet 😅